framework/src/Volo.Abp.Ddd.Domain/Microsoft/Extensions/DependencyInjection/ServiceCollectionRepositoryExtensions.cs

```diff
@@ -80,15 +81,29 @@ public static class ServiceCollectionRepositoryExtensions
         IServiceCollection services,
         Type serviceType,
         Type implementationType,
-        bool replaceExisting)
+        bool replaceExisting,
+        bool isReadOnlyRepository = false)
     {
+        var descriptor = ServiceDescriptor.Transient(serviceType, implementationType);
+
+        if (isReadOnlyRepository)
+        {
+            services.AddTransient(implementationType);
```

Comment:
```suggestion
            services.TryAddTransient(implementationType);
```

Try would be better here.